### DISCUSSION TITLE
Code-Style - Triple Slashed '///' Comments

### DISCUSSION THREAD

**Discussion by kykrueger** [nebulastream/nebulastream#18] [Polls]
In order to make it easier to separate docstrings and important comments from commented code, we would like to consider the adoption of triple slashed comments for docstrings and possibly also important comments.

As we transition to triple slashed comments, it would make it easier to scan the codebase for commented code and dead comments that should be removed.

Edit by @alepping :
If adding an option would not reset the results, I would add and vote for:
- Triple slashed for single-line comments and `/* */` for multi-line comments

**Comment by alepping** [comment]
The way ClickHouse does this is:
- there must not be any comments using `//` (these are just to temporarily comment out code).
- all inline comments use `///` (triple slashed).
- `/* .. */` for multi-line comments.
source: https://clickhouse.com/docs/en/development/style

**Reply by zeuchste** [reply]
but why? what is wrong with //?

**Reply by keyseven123** [reply]
@alepping and @kykrueger have explained the problems with `//`

**Reply by kykrueger** [reply]
I don't really see anything wrong with // in principle. The idea is rather that there is added benefit by using /// instead so that one has a quick extra way to differentiate between temporary comments and ones that should remain for a commit.

**Reply by alepping** [reply]
What @kykrueger said is the exact reason that I had in my mind.
